src/odemis/driver/smaract.py

```diff
@@ -291,119 +295,203 @@ def __str__(self):
+        hold_time (float>=0): the hold time, in seconds, for the actuator after the target position is reached.
+            Default is infinite (float('inf') in Python, .inf in YAML). Can be also set to 0 to disable hold.
         """
         if not axes:
             raise ValueError("Needs at least 1 axis.")
 
-        if locator != "fake":
-            self.core = SmarPodDLL()
-        else:
+        if hold_time < 0:
+            raise ValueError(f"hold_time should be > 0, but got {hold_time}")
+        self._hold_time = hold_time
+
+        if locator == "fake":
             self.core = FakeSmarPodDLL()
+        else:
+            self.core = SmarPodDLL()
             
         # Not to be mistaken with axes which is a simple public view
         self._axis_map = {}  # axis name -> axis number used by controller
         axes_def = {}  # axis name -> Axis object
-        self._locator = c_char_p(locator.encode("ascii"))
-        self._options = c_char_p("".encode("ascii"))  # In the current version, this must be an empty string.
 
-        for axis_name, axis_par in axes.items():
+        for axis_name, param in axes.items():
+            logging.debug("a %s = %s", axis_name, param)
             try:
-                axis_range = axis_par['range']
+                axis_range = param['range']
             except KeyError:
                 logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
                 axis_range = (-1, 1)
 
             try:
-                axis_unit = axis_par['unit']
+                axis_unit = param['unit']
             except KeyError:
                 logging.info("Axis %s has no unit. Assuming m", axis_name)
                 axis_unit = "m"
 
             ad = model.Axis(canAbs=True, unit=axis_unit, range=axis_range)
             axes_def[axis_name] = ad
-            
-            
+
         # Connect to the device
         self._id = c_uint()
-        self.core.Smarpod_Open(byref(self._id), SmarPodDLL.hwModel, self._locator, self._options)
+        try:
+            self.core.Smarpod_Open(byref(self._id), c_ulong(hwmodel),
+                                   c_char_p(locator.encode("ascii")),
+                                   c_char_p(b""))  # last arg is unused
+        except SmarPodError as ex:
+            if ex.errno == SmarPodDLL.NO_SYSTEMS_FOUND_ERROR:
+                raise model.HwError("Failed to find device, check it is connected and turned on") from ex
+            raise
+
         logging.debug("Successfully connected to SmarPod Controller ID %d", self._id.value)
-        self.core.Smarpod_SetSensorMode(self._id, SmarPodDLL.SMARPOD_SENSORS_ENABLED)
+        self.core.Smarpod_SetSensorMode(self._id, SmarPodDLL.SENSORS_ENABLED)
 
         model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
 
         # Add metadata
-        self._swVersion = self.GetSwVersion()
+        # TODO: GetDLLVersion
```

Comment:
What does this TODO mean? Or what needs to be done?

Reply:
Already did it... but forgot to remove the TODO.